pkg/controller/pod.go

```diff
@@ -50,19 +51,21 @@ func (r *ReconcilePod) Reconcile(ctx context.Context, request reconcile.Request)
 		return reconcile.Result{Requeue: true}, nil
 	}
 
+	ctx, workflow := azureutils.NewWorkflow(ctx, consts.PodNameKey, request.Name)
```

Comment:
Missing defer?

Reply:
Same as the comment above. I wonder if we should treat pod controller as a distinct workflow initiator.